libc/time/utimes.c

```diff
@@ -0,0 +1,528 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - sys/time.h
+ *
+ * TESTED:
+ *    - utimes()
+ *    - futimes()
+ *    - lutimes()
+ *
+ * Copyright 2022, 2023 Phoenix Systems
+ * Author: Ziemowit Leszczynski, Mateusz Bloch
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <unistd.h>
+#include <fcntl.h>
+#include <time.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/time.h>
+
+#include "unity_fixture.h"
+
+
+static const char *const FILENAME = "utimes";
+static const char *const LINKNAME = "utimes_link";
+static const unsigned int LOOP_CNT = 10;
+
+
+typedef enum {
+	TEST_SET_NULL,
+	TEST_SET_NOW,
+	TEST_SET_PAST
+} test_t;
+
+
+typedef enum {
+	UTIMES,
+	FUTIMES,
+	LUTIMES
+} mode_test;
+
+
+void test_utimes(const char *filename, test_t test, mode_test mode)
+{
+	int fd;
+	struct timeval tv[2];
+	struct stat statbuf;
+	int64_t adiff_sec, mdiff_sec;
+/* FIXME: only second accuracy is supported */
+#ifndef __phoenix__
+	int64_t adiff_nsec, mdiff_nsec;
+#endif
+
+	if (test != TEST_SET_PAST) {
```

Comment:
I see that you have added new cases - I think that this approach with one common function is a good solution - we can add a `switch case` here to not duplicate code from `one` and `high_value` cases.